### ISSUE TITLE
Allow custom config directories for `NoEnvOutsideConfig`

### ISSUE THREAD

**Issue by alies-dev** [psalm/laravel-psalm-plugin#858]
## Problem

`NoEnvOutsideConfigHandler` hard-codes the config directory detection via a case-insensitive `str_contains` on any path segment named `config`. This has two gaps:

1. **False negatives** — any path containing `/config/` anywhere (e.g. a vendor package's own `config/`) suppresses the issue, even when the `env()` call is genuinely outside the project's config.
2. **False positives** — apps that store config files outside the standard `config/` directory (e.g. BookStack's `app/Config/`) get flagged even though those files are legitimate config files.

## Proposed solution

Add a `configDirectories` XML option (Option A — flat repeating elements) so users can declare the exact config paths for their project.

### XML syntax

```xml
<pluginClass class="Psalm\LaravelPlugin\Plugin">
    <configDirectory name="app/Config" />
    <configDirectory name="packages/*/config" />
</pluginClass>
```

### Default behaviour

When no `<configDirectory>` elements are present, fall back to `config_path()` (the project's Laravel application config path, resolved via `ApplicationProvider::getApp()->configPath()`). This mirrors [Larastan's approach](https://github.com/larastan/larastan/blob/master/src/Rules/NoEnvCallsOutsideOfConfigRule.php).

The default changes the current `str_contains` heuristic to a precise `str_starts_with` check against the actual config directory — a stricter and more correct signal.

### Implementation sketch

**`PluginConfig`** — parse the list:

```php
/** @return list<string> */
private static function xmlStringList(?\SimpleXMLElement $config, string $element): array
{
    $result = [];
    foreach ($config?->{$element} ?? [] as $node) {
        $value = (string) ($node['name'] ?? '');
        if ($value !== '') {
            $result[] = $value;
        }
    }
    return $result;
}
```

Add `public readonly array $configDirectories` to `PluginConfig` and populate it:

```php
$configDirectories = self::xmlStringList($config, 'configDirectory');
// if empty, resolved later from config_path()
```

**`NoEnvOutsideConfigHandler`** — replace the current `str_contains` with `str_starts_with` against the resolved directories (with glob + `is_dir` guard, same as Larastan):

```php
private static function isInsideConfigDirectory(string $filePath): bool
{
    foreach (self::$configDirectories as $glob) {
        foreach (glob($glob) ?: [] as $dir) {
            if (is_dir($dir) && str_starts_with($filePath, realpath($dir) . DIRECTORY_SEPARATOR)) {
                return true;
            }
        }
    }
    return false;
}
```

**`Plugin::__invoke()`** — wire up:

```php
$dirs = $pluginConfig->configDirectories ?: [ApplicationProvider::getApp()->configPath()];
NoEnvOutsideConfigHandler::init($dirs);
```

### Acceptance criteria

- Default (no XML option): `config_path()` is used; files under that directory are suppressed.
- Custom directories: user-specified paths (including glob patterns) override the default.
- BookStack `app/Config/` scenario: adding `<configDirectory name="app/Config" />` silences the false positives.
- Existing unit tests pass; new tests cover the XML parsing and the default fallback.
